core/container.go

```diff
@@ -476,15 +527,32 @@ func (container *Container) Exec(ctx context.Context, gw bkgw.Client, args *[]st
 		runOpts = append(runOpts, llb.AddEnv(name, val))
 	}
 
-	st, err := payload.FSState()
+	for i, secret := range payload.Secrets {
+		secretOpts := []llb.SecretOption{llb.SecretID(string(secret.Secret))}
+
+		var secretDest string
+		switch {
+		case secret.EnvName != "":
+			secretDest = secret.EnvName
+			secretOpts = append(secretOpts, llb.SecretAsEnv(true))
+		case secret.MountPath != "":
+			secretDest = secret.MountPath
+		default:
+			return nil, fmt.Errorf("malformed secret config at index %d", i)
+		}
+
+		runOpts = append(runOpts, llb.AddSecret(secretDest, secretOpts...))
```

Comment:
nit: followup issue for some of the other options that users may need for secret mounts (mode, uid gid)

Reply:
Opened https://github.com/dagger/dagger/issues/3323